openmdao/core/group.py

```diff
@@ -3371,7 +3371,7 @@ def _get_auto_ivc_out_val(self, tgts, vars_to_gather, abs2meta_in):
                     if val.size > 1:
                         raise ValueError(f"Shape of input '{tgt}', (), doesn't match shape "
                                          f"{val.shape}.")
-                elif val.shape != value.shape:
+                elif val.shape != value.shape and val.size > 1 and value.size > 1:
                     raise ValueError(f"Shape of input '{tgt}', {value.shape}, doesn't match shape "
                                      f"{val.shape}.")
                 if val is not value:
```

Comment:
Instead of doing this, what would you think about checking the "squeezed" shape (np.squeeze), so that inputs with shapes (1, 3, 1) and (3, ) could also be connected together?

Reply:
Good idea, I'll make that change.